packages/pwa-kit-extension-sdk/src/configs/webpack/application-extensions-loader.ts

```diff
@@ -37,7 +38,10 @@ export default function ApplicationExtensibilityLoader(
     // For web targets, the loader takes advantage of react-loadable but node targets (server) do not require this optimization.
     const data = this.getOptions()
 
-    return renderTemplate(data)
+    return renderTemplate({
+        ...data,
+        configured: expand(data.configured).map((extension) => mergeConfigs(extension))
+    })
```

Comment:
This is where the extension-sdk handles merging of the configs. Right before rendering the handlebar template. In the webpack loader and also similarly in the babel plugin.